File: src/com/javascript-practice/linkList/longestStreak.js


const longestStreak = (node) => {
    let max = 0;
    if(node == null)  return max;
    let firstVal = null;
    let count = 0;
    let current  = node;
    while(current!=null) {
        if(firstVal == current.val) {
            firstVal = current.val;
            count ++;
        }else {
            count = 1;
            firstVal = current.val;
        }
        if(count > max) {
            max = count;
        }
        current = current.next;
    }
    return max;
    
}




class Node {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}
const a = new Node(5);
const b = new Node(5);
// const c = new Node(7);
// const d = new Node(7);
// const e = new Node(7);
// const f = new Node(6);

a.next = b;
// b.next = c;
// c.next = d;
// d.next = e;
// e.next = f;

// 5 -> 5 -> 7 -> 7 -> 7 -> 6

let result = longestStreak(a); // 2
console.log(result);

File: src/com/structy/LinkedList/LongestStreak.java
package com.structy.LinkedList;

public class LongestStreak {
    public int longestStreak(Node head){
        int maxStreak = 0 ;
        if(head == null) return maxStreak;
        int currentStreak  = 0;
        int firstVal = Integer.MIN_VALUE;
        Node current = head;

        while (current!=null){
            if(firstVal == current.val){
                firstVal = current.val;
                currentStreak++;
            }else {
                currentStreak = 1;
                firstVal = current.val;
            }
            maxStreak = Math.max(maxStreak,currentStreak);
            current = current.next;
        }
        return maxStreak;
    }

    public static void main(String[] args) {
        Node a = new Node(5);
        Node b = new Node(5);
//        Node c = new Node(7);
//        Node d = new Node(7);
//        Node e = new Node(7);
//        Node f = new Node(6);
        a.next = b;
//        b.next = c;
//        c.next = d;
//        d.next = e;
//        e.next = f;
        LongestStreak longestStreak = new LongestStreak();
        System.out.println(longestStreak.longestStreak(a));
    }
}
